iosHyperskillApp/iosHyperskillApp/Sources/Modules/StudyPlan/Views/StudyPlanView.swift

```diff
@@ -0,0 +1,121 @@
+import shared
+import SwiftUI
+
+extension StudyPlanView {
+    struct Appearance {
+        let backgroundColor = Color(ColorPalette.background)
+
+        let subheadlineBottomPadding: CGFloat = 12
+    }
+}
+
+struct StudyPlanView: View {
+    private(set) var appearance = Appearance()
+
+    @StateObject var viewModel: StudyPlanViewModel
+
+    @StateObject var stackRouter: SwiftUIStackRouter
+
+    var body: some View {
+        ZStack {
+            UIViewControllerEventsWrapper(onViewDidAppear: viewModel.logViewedEvent)
+
+            BackgroundView(color: appearance.backgroundColor)
+
+            buildBody()
+        }
+        .navigationTitle(Strings.StudyPlan.title)
+        .navigationViewStyle(StackNavigationViewStyle())
+        .toolbar {
+            GamificationToolbarContent(
+                stateKs: viewModel.gamificationToolbarStateKs,
+                onGemsTap: viewModel.doGemsBarButtonItemAction,
+                onStreakTap: viewModel.doStreakBarButtonItemAction
+            )
+        }
+        .onAppear {
+            viewModel.startListening()
+            viewModel.onViewAction = handleViewAction(_:)
+        }
+        .onDisappear {
+            viewModel.stopListening()
+            viewModel.onViewAction = nil
+        }
+    }
+
+    // MARK: Private API
+
+    @ViewBuilder
+    private func buildBody() -> some View {
+        switch viewModel.studyPlanWidgetStateKs {
+        case .idle:
+            StudyPlanSkeletonView()
+                .onAppear {
+                    viewModel.doLoadStudyPlan()
+                }
+        case .loading:
+            StudyPlanSkeletonView()
+        case .error:
+            PlaceholderView(
+                configuration: .networkError(
+                    backgroundColor: appearance.backgroundColor,
+                    action: viewModel.doRetryContentLoading
+                )
+            )
+        case .content(let data):
+            ScrollView(showsIndicators: false) {
+                VStack(alignment: .leading, spacing: LayoutInsets.defaultInset) {
+                    if let trackTitle = viewModel.state.trackTitle {
+                        Text(trackTitle)
+                            .font(.subheadline)
+                            .foregroundColor(.secondaryText)
+                            .padding(.bottom, appearance.subheadlineBottomPadding)
+                    }
+
+                    ProblemsLimitAssembly().makeModule()
+
+                    ForEach(data.sections, id: \.id) { section in
+                        StudyPlanSectionView(
+                            section: section,
+                            onSectionTap: { viewModel.doSectionToggle(sectionId: section.id) },
+                            onActivityTap: viewModel.doActivityPresentation(activityId:)
+                        )
+                    }
+                }
+            }
```

Comment:
By default, SwiftUI’s `VStack` and `HStack` load all their contents up front, which is likely to be slow when used inside a scroll view. It's better to load content lazily – i.e., only when it scrolls into view, you should use `LazyVStack` and `LazyHStack` as appropriate.

`ScrollView` + `LazyVStack` - when using this SwiftUI will automatically create a view when it gets shown for the first time. After that the view will remain in memory, so we will need to be careful how much we show.

If we want to reuse cells and be more respectful about memory usage we should use `List`.

For now, I think it would be OK to use lazy stacks, we will monitor memory consumption with a study plan and make appropriate changes if needed.